mesonbuild/environment.py

```diff
@@ -628,6 +647,54 @@ def move_compiler_options(properties: Properties, compopts: T.Dict[str, T.Defaul
                     del properties.properties[k]
                     break
 
+        def read_env_vars(for_machine: MachineChoice) -> None:
+            for key, envkey in OTHER_OPTS_TO_ENV.items():
+                p_env_pair = get_env_var_pair(for_machine, self.coredata.is_cross_build(), envkey)
+                if p_env_pair is not None:
+                    p_env_var, p_env = p_env_pair
+
+                    if mesonlib.is_windows():
+                        splitter = lambda p: p.split(';')
+                    else:
+                        splitter = lambda p: re.split(r':|;', p)
```

Comment:
There was a link to a bug in the code you deleted above regarding that split. I haven't read what it's about, but I guess it's better to keep it. I find it suspect that we split on both chars, ideally it should be only os.pathsep.

Reply:
Specifically in CMake `;` is used as a list separator on all platforms since this is a core language "feature". So we have to expect that all env vars and config options related to CMake can use `;` as a list separator.

Reply:
Ok, worth adding a comment there mentioning cmake then.